web/src/components/software/PatternSelector.jsx

```diff
@@ -166,6 +166,12 @@ function PatternSelector({ patterns, onSelectionChanged = noop }) {
     );
   });
 
+  if (selector.length === 0) {
+    selector = (
+      <b>{_("None of the patterns match the text.")}</b>
+    );
+  }
```

Comment:
NP:

Sorry, I didn't realize before, but it is actually not needed to redefine the selector variable. You can just change `{selector}` to `{selector.length ? selector : <NoResultFoundThing />}` below. But don't worry ant keep it as it is to minimize the conflicts with the new-ui branch.